client/mainmenu/CPrologEpilogVideo.cpp

```diff
@@ -31,32 +33,42 @@ CPrologEpilogVideo::CPrologEpilogVideo(CampaignScenarioPrologEpilog _spe, std::f
 	videoSoundHandle = CCS->soundh->playSound(audioData);
 	CCS->videoh->open(spe.prologVideo);
 	CCS->musich->playMusic(spe.prologMusic, true, true);
+	voiceDuration = CCS->soundh->getSoundDuration(spe.prologVoice);
 	voiceSoundHandle = CCS->soundh->playSound(spe.prologVoice);
 	auto onVoiceStop = [this]()
 	{
 		voiceStopped = true;
+		elapsedTime = 0.0;
 	};
 	CCS->soundh->setCallback(voiceSoundHandle, onVoiceStop);
 
 	text = std::make_shared<CMultiLineLabel>(Rect(100, 500, 600, 100), EFonts::FONT_BIG, ETextAlignment::CENTER, Colors::METALLIC_GOLD, spe.prologText.toString());
-	text->scrollTextTo(-100);
+	text->scrollTextTo(-50);
 }
 
 void CPrologEpilogVideo::show(Canvas & to)
 {
+	elapsedTime += GH.framerate().getElapsedMilliseconds() / 1000.0;
```

Comment:
Instead of querying framerate this code should be in `void tick(uint32_t msPassed)` method